master/buildbot/process/factory.py

```diff
@@ -92,19 +92,13 @@ def __init__(self, source, configure="./configure",
                  configureEnv=None,
                  configureFlags=None,
                  reconf=None,
-                 compile=None,
-                 test=None,
-                 distcheck=None):
+                 compile=["make", "all"],
```

Comment:
Not sure about setting the steps with list values because of how Python handles it. Since the docs are already wrong, I think it would be better to use `False` to disable them.